### DISCUSSION TITLE
Bots That Create Accounts...

### DISCUSSION THREAD

**Discussion by omicreativedev** [deso-protocol/dips#65] [Q&A]
So as it turned out, someone used a bot to make just under 300 clone accounts of Twitter accounts.
Something suggested by Brootle was that we modify the code so that notifications don't trigger when a users coin is 0.
I think that's really a great idea.
At present, you can't update your profile unless you have something in your wallet for gas.
I think it would make sense that if the coin price is 0, that you can't LIKE or DM.
You can still comment, post, reclout, diamond if you have the gas to do so.
Maybe we could also consider adding CAPSHA to nodes sign up on the frontend.
It could be turned off or key config on a node by node basis, but at least we'd see what node has capsha off and is being botted.

We should not remove the like feature because there are so many important dapps that can be built on bitclout using it, and also on layers 3, 4, etc.

Its a great feature to be able to automate account creation for legitimate reasons- such as when other legitimate networks want to add a node and sync a database for instance. 

It wasn't that many new accounts that botted in. I'm sure it's easily swapped. I'm working on a spreadsheet and contacting a few hundred to see if any of those in the time window are real.

Can we brainstorm on ways we can discourage account creation botting without removing "like" and without depending on swap identity to fix? Without risking the real accounts that joined in the same window?

**Comment by tijno** [comment]
This is not the first time these specific bots hit. Turns out they were doing this last week at a slightly lower scale.

They were definitely Bots though. Liking a reclout posthash that you cant like in the UI.

As always, going to come at this from the alternative side. 

Doesnt mean I disagree - but i think its important to  challenge ourselves and think broadly.

The challenge with putting in place rules on the blockchain/consensus/peotocol about what accounts are or are not allowed to do ... i think is a slippery slope.

Ok so we block likes / DMs / follows at 0 coin value.

What undesirable outcomes & consequences will this trigger ?

Eg will bots try and get investment, or will they just invest a little to get past the "barier".

What about real users - they have a hard enough time.

So next time we increase it.

And then we see more undesirable behaviour so we add more rules to the protocol.

eg Will we stop accounts selling their own coins unless they meet standard xyz.

Now there is a valid case of course to stop actions that precent the protocol from functioning.

Bitcoin has had to solve this with certain spam transactions and they indeed implemented some rules in code. 

But i dont think this like/follow action falls in that category.

What solutions are there that dont require protocol changes ?

**Reply by omicreativedev** [reply]
I actually agree with you that it would be BETTER not have to make protocol changes.
Even capsha does nothing a lot of times, because there are botnets that set up accounts 1 by 1.
You know, the "Buy Youtube Likes" services. No big deal to me personally. I'm not phased personally. 
When I was on Steemit, botting was a problem but their approach was very firm. 
Being firm definately pissed off people who were making good tools for good reasons.
That's the sucky thing too. I love voluntary curation trails with autoliking with bots on some networks. It would utilize similar mechanism. But then you get people who just do it to increase server load and slow down hashing, then it's a problem.

**Reply by matt-mcmarsh** [reply]
Hi
(not all that familiar with github)
but wanted to suggest one "fix" for these bot's notifications
would be filter by coin price on notifications (and also a filter users by number of followers as addition)
at least then I could show only notifications from accounts with coin higher than x $ if I could filter them
at the moment during attack on my posts my notifications become useless
and I'm probably going to give up posting for the rest of the night as I don't have the energy to bother

**Comment by sull** [comment]
def a frontend tweak. either user-defined config in settings if not only by node admin config.

too granular and tricky for protocol level.

**Comment by brootle** [comment]
@bluepartyhat don't add notifications from accounts that have coin price 0.00, easy fix

**Comment by matt-mcmarsh** [comment]
Very quick mock up
![notifications3](https://user-images.githubusercontent.com/83852756/125924746-a3b66590-0d76-4b47-b7bc-5f96e7697513.png)

**Reply by matt-mcmarsh** [reply]
It is more an improvement to notifications
but would also allow to avoid the annoyance of the bot notifications

**Reply by tijno** [reply]
A major rewrite of notifications is underway i believe 

https://github.com/orgs/bitclout/projects/1#card-64521042

So maybe in short term just a "hide $0 notifications" checkbox and then add the more customisable options later on ones notifications rewritten?
